m4/ax_lua.m4

```diff
@@ -49,6 +49,14 @@
 #   interpreter. If LUA is blank, the user's path is searched for an
 #   suitable interpreter.
 #
+#   Optionally a LUAJIT option may be set ahead of time to look for and
+#   validate a LuaJIT install instead of PUC Lua. Usage might look like:
+#
+#     AC_ARG_WITH(luajit, [AS_HELP_STRING([--with-luajit],
+#         [Prefer LuaJIT over PUC Lua, even if the latter is newer. Default: no])
+#     ])
+#     AM_CONDITIONAL([LUAJIT], [test "x$with_luajit" != 'xno'])
```

Comment:
> This implementation leaves it up to the user to specifically request LuaJIT be considered by adding a `--with-luajit` configure option. If they do that and it is set to true, then the regular Lua macros will function in the usual way but with the binary names, include paths, and other things tweaked for use with LuaJIT.

This doesn't seem to be very optional at all: https://bugs.gentoo.org/941718

Per the automake documentation:

> Before using a conditional, you must define it by using AM_CONDITIONAL in the configure.ac file (see [Autoconf macros supplied with Automake](https://www.gnu.org/software/automake/manual/html_node/Macros.html)).

So any previously existing users of ax_lua.m4 now get a fatal autoreconf error if they don't support luajit, which appears to be precisely the opposite of what you wanted. :(

```
configure.ac:641: error: AM_COND_IF: no such condition "LUAJIT"
/usr/share/aclocal-1.17/cond-if.m4:23: AM_COND_IF is expanded from...
/usr/share/aclocal/ax_lua.m4:199: AX_PROG_LUA is expanded from...
```

Reply:
Yes, my bad. All my test projects had this conditional set one way or the other. I tested with default on and default off and using config flag overrides both ways, but apparently not without doing anything at all. See #313 for a fix.